openapi3filter/req_resp_decoder.go

```diff
@@ -1221,3 +1224,44 @@ func FileBodyDecoder(body io.Reader, header http.Header, schema *openapi3.Schema
 	}
 	return string(data), nil
 }
+
+// ZipFileBodyDecoder is a body decoder that decodes a zip file body to a string.
+func ZipFileBodyDecoder(body io.Reader, header http.Header, schema *openapi3.SchemaRef, encFn EncodingFn) (interface{}, error) {
+	buff := bytes.NewBuffer([]byte{})
+	size, err := io.Copy(buff, body)
+	if err != nil {
+		return nil, err
+	}
+
+	zr, err := zip.NewReader(bytes.NewReader(buff.Bytes()), size)
+	if err != nil {
+		return nil, err
+	}
+
+	const bufferSize = 256
+	var content []byte
+	buffer := make([]byte, bufferSize)
+
+	for _, f := range zr.File {
+		rc, err := f.Open()
+		if err != nil {
+			return nil, err
+		}
+		defer rc.Close()
```

Comment:
Wrap the contents of this for loop inside a `func(){ ... ; return }()` as `defer` happens only at the next `return` statement.

Reply:
Or don't use `defer`

Reply:
Does it look like this?
https://github.com/getkin/kin-openapi/pull/730/commits/5361b9b8b7cd549e34b13409f6dd919079e89b3a